CMakeLists.txt

```diff
@@ -175,6 +175,7 @@ include(external/eigen)     # download eigen3
 include(external/pybind11)  # download pybind11
 include(external/cares)
 include(external/cub)
+include(external/jemalloc)
```

Comment:
Maybe ```WITH_JE_MALLOC``` could be a ```cmake``` flags and the allocator could be determined compile time.

Then we could use ```WITH_TC_MALLOC``` and ```WITH_BUDDY_ALLOCATOR``` to change it